libkmod/libkmod.c

```diff
@@ -564,15 +585,22 @@ int kmod_lookup_alias_from_config(struct kmod_ctx *ctx, const char *name,
 				goto fail;
 			}
 
-			*list = kmod_list_append(*list, mod);
+			node = kmod_list_append(*list, mod);
+			if (node == NULL) {
+				ERR(ctx, "out of memory\n");
+				kmod_module_unref(mod);
+				err = -ENOMEM;
+				goto fail;
+			}
+			*list = node;
 			nmatch++;
 		}
 	}
 
 	return nmatch;
 
 fail:
-	*list = kmod_list_remove_n_latest(*list, nmatch);
+	kmod_list_release(*list, kmod_module_unref);
```

Comment:
this doesn't seem correct.  It doesn't seem we mandate the list to be empty, so we may be releasing something this function didn't really append.

Reply:
if we double check every caller, then we'd need to add an assert() in the beginning of the function.

Reply:
It is correct, because these functions are eventually called from `kmod_module_new_from_lookup`. Adding `assert` sounds reasonable.